tests/test_client_request.py

```diff
@@ -1230,3 +1230,61 @@ def test_loose_cookies_types(loop: Any) -> None:
 def test_gen_default_accept_encoding(has_brotli: Any, expected: Any) -> None:
     with mock.patch("aiohttp.client_reqrep.HAS_BROTLI", has_brotli):
         assert _gen_default_accept_encoding() == expected
+
+
+@pytest.mark.parametrize(
+    ["netrc_contents", "hostname", "trust_env", "expected_auth"],
+    [
+        (
+            "machine example.com login username password pass\n",
+            "example.com",
+            True,
+            helpers.BasicAuth("username", "pass"),
+        ),
+        (
+            "machine example.com login username password pass\n",
+            "example.com",
+            False,
+            None,
+        ),
+    ],
+    indirect=("netrc_contents",),
+)
```

Comment:
This can be simplified a bit
```suggestion
@pytest.mark.parametrize(
    ("trust_env", "expected_auth"),
    (
        pytest.param(
            True,
            helpers.BasicAuth("username", "pass"),
            id='reading netrc requested',
        ),
        pytest.param(
            False,
            None,
            id='reading netrc disabled',
        ),
    ),
)
@pytest.mark.parametrize(
    "netrc_contents",
    (
        "machine example.com login username password pass\n",
    ),
    indirect=("netrc_contents",),
)
```